app/Http/Controllers/AlbumController.php

```diff
@@ -384,14 +392,25 @@ function merge(Request $request)
 		$no_error = true;
 		foreach ($photos as $photo) {
 			$photo->album_id = $albumID;
+			if ($this->sessionFunctions->is_admin()) {
+				// Admin can merge albums between users.  Make sure that the
+				// ownership changes in the process.
+				$photo->owner_id = $album->owner_id;
+			}
 			$no_error &= $photo->save();
 		}
 
 		$albums = Album::whereIn('parent_id', $albumIDs)->get();
 		$no_error = true;
-		foreach ($albums as $album) {
-			$album->parent_id = $albumID;
-			$no_error &= $album->save();
+		foreach ($albums as $album_t) {
+			$album_t->parent_id = $albumID;
+			if ($this->sessionFunctions->is_admin()) {
```

Comment:
idem as above. :)
The only gain you have is a slight code speed up for non-admin users.

Reply:
The reason I did it like that was that in some cases the Album object was not readily available and needed to be looked up. So I didn't want to slow down the common case (a regular user moving their own objects around) to handle a corner case. But I take your point that it was making the code look more complicated.

Reply:
:)